qgym/benchmarks/metrics/initial_mapping_metrics.py

```diff
@@ -0,0 +1,146 @@
+            return 1.0
+        mapping = np.asarray(mapping, dtype=np.int_)
+        distance_loss = 0
+        for edge in interaction_graph.edges():
+            mapped_edge = (mapping[edge[0]], mapping[edge[1]])
+            distance_loss += self.distance_matrix[mapped_edge]
+
+        return float(distance_loss / interaction_graph.number_of_edges())
+
+
+class AgentPerformance:
+
+    def __init__(
+        self,
+    ) -> None:
+        """Init of the :class:`AgentPerformance` class.
+
+        Args:
+        """
+
+
+class InitialMappingBenchmarker:
+
+    def __init__(
+        self,
+        generator: GraphGenerator | None = None,
+        *,
+        metrics: Iterable[InitialMappingMetric] | InitialMappingMetric,
+    ) -> None:
+        """Init of the :class:`InitialMappingBenchmarker` class.
+
+        Args:
+            generator: Interaction graph generator to use during benchmarking
+            metrics: Metrics to compute.
+        """
+        self.generator = BasicGraphGenerator() if generator is None else generator
+        self.metrics = (
+            (metrics,) if isinstance(metrics, InitialMappingMetric) else tuple(metrics)
+        )
+        for metric in self.metrics:
+            if hasattr(metric, "connection_graph"):
+                connection_graph = deepcopy(metric.connection_graph)
+                self.generator.set_state_attributes(connection_graph=connection_graph)
+                break
+
+    def run(
+        self,
+        mapper: Mapper,
+        max_iter: int = 1000,
+        return_type: str = "raw",
+    ) -> NDArray[np.float64]:
+        """Run the benchmark.
+
+        Args:
+            mapper: Mapper to benchmark.
+            max_iter: Maximum number of iterations to benchmark.
+            return_type: Return type to return the results. Choose from ["raw",
+                "quartiles", "median", "mean"]
+
+        Returns:
+            NDArray containing the results from the benchmark.
+        """
+        return_type = check_string(return_type, "return_type", lower=True)
+
+        results: list[deque[float]] = [deque() for _ in self.metrics]
+        for i, interaction_graph in enumerate(self.generator, start=1):
+            mapping = mapper.compute_mapping(interaction_graph)
+            for metric, result_que in zip(self.metrics, results):
+                result_que.append(metric.compute(interaction_graph, mapping))
+
+            if i >= max_iter:
+                break
+
+        if return_type == "raw":
+            return cast(NDArray[np.float64], np.array(results, dtype=np.float64))
+        if return_type == "quartiles":
+            return cast(
+                NDArray[np.float64],
+                np.quantile(results, [0, 0.25, 0.5, 0.75, 1], axis=1),
+            )
+        if return_type == "median":
+            return cast(NDArray[np.float64], np.median(results, axis=1))
+        if return_type == "mean":
+            return cast(NDArray[np.float64], np.mean(results, axis=1))
```

Comment:
What about having a ReturnType, or RunResult, or BenchmarkResult:

- which holds internally this `NDArray[np.float64]`, and
- offers a series of `as_raw`, `as_quartiles`, `as_median`, `as_mean` API?

So the caller wouldn't do `run(..., "quartiles")`, but `run(...).as_quartiles()`.

Let's say this code shouldn't belong here, even if it's the only point in the code where it is used:

- `run` would receive a fewer parameter, wouldn't check that parameter, and would return a result.
- For  the caller the code wouldn't be much more complicated. They would understand that `run` just returns a result and that the Result type offers different casts.
- Arguably, the casts wouldn't be either part of the Result type, they should just be part of the language. But they are complex lines, very much focus in the Result type, and we can consider them helpers.